schemas/schemas.go

```diff
@@ -1,8 +1,99 @@
 package schemas
 
-// MessagePacket is the low-level format for storing contents of a message and arguments
-type MessagePacket struct {
-	Data string `json:"data" validate:"required"`
-	Text string `json:"text" validate:"required"`
-	Type string `json:"type" validate:"required"`
+import (
+	"strings"
+)
+
+type Message struct {
+	Text   []Component `json:"text"` // FIXME: Anything besides [] breaks this. It is assumed that this is due to the interface being used for Component.
```

Comment:
@RPiAwesomeness This is critical to fix before merging; we obviously need to be able to `PATCH` new commands! Any ideas?